homeassistant/components/komfovent/config_flow.py

```diff
@@ -0,0 +1,73 @@
+"""Config flow for Komfovent integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import komfovent_api
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, PARAM_HOST, PARAM_PASSWORD, PARAM_USERNAME
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER = "user"
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(PARAM_HOST): str,
+        vol.Required(PARAM_USERNAME, default="user"): str,
```

Comment:
Wouldn't it still be required? You can't complete a setup without it, and even though it has a default, a user can delete the text from the textbox?

Reply:
If it is optional, and not provided, it will use the default provided.
Setting a default on required, however, won't have much use (as it won't accept an empty input).